session_4/blog/app/controllers/articles_controller.rb

```diff
@@ -102,4 +125,23 @@ def set_article
     def article_params
       params.require(:article).permit(:title, :topic, :tags, :content, :public, :image)
     end
+    
+    #to add views
+    def add_view
+      if current_user
+        if !View.find_by("user_id = ? AND article_id = ? ", current_user.id, @article.id )
```

Comment:
Use `exists?(user_id: current_id, article_id: @article.id)` instead of writing SQL syntax by hand

Reply:
I will change that.
Is their a different way of writing this `Article.where(" views > ? AND likes > ?")` .

Reply:
You can try `Article.where(views: 200..Float::Infinity, likes: 100..Float::Infinity)` although these should be two scopes - `with_views_more_than(x)` and `with_likes_more_than(x)`

https://stackoverflow.com/a/23936233

https://www.rubyguides.com/2019/10/scopes-in-ruby-on-rails/

Reply:
Thank you for the help